.github/workflows/publish-release.yml

```diff
@@ -323,3 +485,40 @@ jobs:
               run: |
                   security delete-keychain $RUNNER_TEMP/app-signing.keychain-db
                   rm -f ~/Library/MobileDevice/Provisioning\ Profiles/build_pp.mobileprovision
+
+    release_web_checksums:
+        name: "Release web checksums"
+        environment: ${{ github.event.inputs.environment || 'Beta' }}
+        needs: create_release_tag
+        env:
+            RELEASE_ID: ${{ needs.create_release_tag.outputs.release_id }}
+            PL_VENDOR_VERSION: ${{ github.event.inputs.vendor_version || '0.0.1' }}
+
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v2
+              with:
+                  node-version-file: ".nvmrc"
+            - name: Install dependencies
+              run: |
+                  npm i -g npm@8.2.0
+                  npm ci
+            - name: Build
+              run: |
+                  npm run pwa:build
+              env:
+                  PL_PWA_URL: ${{ secrets.PL_PWA_URL }}
+                  PL_SERVER_URL: ${{ secrets.PL_SERVER_URL }}
+            - name: Generate checksums
+              run: |
+                  cd packages/pwa/dist
+                  find . -type f ! \( -name "*.txt" -or -name "*.png" -or -name "*.json" -or -name "*.svg" -or -name "*.woff2" -or -name "*.map" \) -exec sha256sum {} > sha256sums-web.txt \;
```

Comment:
I don't see `*.js`? Am I missing something?

Reply:
Ah, ok, missed the `not`. Which leads to to a different question: Why exclude those files? Any of those could potentially the source of a vuln if compromised (unlikely, but you never know...). So while we're at it, why not check everything?

Reply:
I'm excluding them because they're not relevant from a security point of view and they're prone to being changed from system to system (imagine someone's using this documentation to check on some Padloc install other than Padloc's web app). I could've instead just included *.js and *.html indeed. Also I needed to exclude the `sha256sums-web` file itself otherwise running the command locally would fail on itself.

Reply:
> because they're not relevant from a security point of view

I would disagree. `json` and `text` files can be read dynamically from the app, and *could* change certain behaviours.  `.map` files can be used to obfuscate the actual source when viewed trough the dev tools. Even things like `png` files [have been known](https://www.mozilla.org/en-US/security/advisories/mfsa2010-41/) to be used as attack vectors. 
So unless there is a very good reason against it, I would simply include all of them.

> they're prone to being changed from system to system

Really? Which ones and how/why?

Reply:
> So unless there is a very good reason against it, I would simply include all of them.

I understand, but they're not allowed via CSP, and since they're not included in the CSP, we can't really get their list from looking at a web app. Also, since they're not in the CSP, they can't be loaded, even dynamically.

> Really? Which ones and how/why?

`robots.txt` will vary from host to host, and likely the `png` and `svg` files as well. Some people might have slightly different fonts too, and any small change will generate a completely different filename hash and contents, which will fail the checksum.

Reply:
> I understand, but they're not allowed via CSP

Pretty sure all of these file types (with the possible exception of the source maps) can (and maybe should) be whitelisted explicitly through the CSP (`font-src` for fonts, `img-src` for pngs etc.).

> robots.txt will vary from host to host

Where do we have a `robots.txt`? 

> and likely the png and svg files as well. Some people might have slightly different fonts too, and any small change will generate a completely different filename hash and contents, which will fail the checksum.

I don't understand - they're included with the source code so why should the vary? Are you talking about white-labeling scenarios?

Reply:
> Pretty sure all of these file types (with the possible exception of the source maps) can (and maybe should) be whitelisted explicitly through the CSP

Ok, I can appreciate that plan, even if it makes this take a bit longer.

> Where do we have a `robots.txt`?

When you try to download a website with `wget -r -p` you'll get "common" files like `robots.txt` that are generally expected, and our beta environment exposes that (any deployment should, really, just to make sure things are/aren't indexed properly).

> I don't understand - they're included with the source code so why should the vary? Are you talking about white-labeling scenarios?

I'm talking about scenarios where people might install on their own server and likely change the logo, among other small customizations. If you don't want to cater to those scenarios, then it's fine to include them in the checks, and let people judge if/when it's fine if they're not matching the checksums, I assume.

Reply:
> When you try to download a website with wget -r -p you'll get "common" files like robots.txt that are generally expected, and our beta environment exposes that (any deployment should, really, just to make sure things are/aren't indexed properly).

But we're only checking the files that are included in the CSP, so we don't have to worry about those extraneous files, right?

> I'm talking about scenarios where people might install on their own server and likely change the logo, among other small customizations. If you don't want to cater to those scenarios, then it's fine to include them in the checks, and let people judge if/when it's fine if they're not matching the checksums, I assume.

Why would people check their custom builds against the official checksums? If they're making customisations, then of course the checksums won't match - that's the whole point 😊. If people create custom builds and want those to be verifiable by others, they should publish their own checksums.

Reply:
> But we're only checking the files that are included in the CSP, so we don't have to worry about those extraneous files, right?

We weren't, but we can. These files will also be downloaded by default, so I figured it would make sense to get them, as it would be the "default browser behavior" we talked about trying to emulate before. I'm fine ignoring them.

> Why would people check their custom builds against the official checksums?

If they said "the code is the same, trust me, I just changed the images and fonts", they could actually check that and it would be fine/safe. I'll tweak all of this stuff then.

Reply:
> If they said "the code is the same, trust me, I just changed the images and fonts", they could actually check that and it would be fine/safe.

I don't think that's a use case we want to support. With this stuff it's generally better to err on the side of more strictness.

Reply:
> We weren't, but we can. These files will also be downloaded by default, so I figured it would make sense to get them, as it would be the "default browser behavior" we talked about trying to emulate before. I'm fine ignoring them.

If they're not included in the CSP, they can't/won't be loaded in the context of the PWA, so these are actually irrelevant.

Reply:
> I don't think you can prevent a browser from trying to get robots.txt and favicon.ico, for example, but that's ok.

Pretty sure the CSP will prevent those from loading as well if blocked / not whitelisted.

Reply:
Not `robots.txt`. `favicon.ico` _maybe_. It doesn't matter here, anyway.